src/librustc/mir/interpret/error.rs

```diff
@@ -309,8 +309,30 @@ impl<O: fmt::Debug> fmt::Debug for PanicMessage<O> {
     }
 }
 
+#[derive(Clone, RustcEncodable, RustcDecodable, HashStable)]
+pub enum InvalidProgramMessage<'tcx> {
```

Comment:
Seems like you can remove the `'tcx` here. I had it everywhere in my post just because I did not know where it was needed.